docs/book/src/developer/tilt.md

```diff
@@ -35,7 +35,8 @@ Next, create a `tilt-settings.json` file and place it in your local copy of `clu
   "allowed_contexts": ["kind-kind"],
   "default_registry": "gcr.io/your-project-name-here",
   "provider_repos": ["../cluster-api-provider-aws"],
-  "enable_providers": ["aws", "docker", "kubeadm-bootstrap", "kubeadm-control-plane"]
+  "enable_providers": ["aws", "docker", "kubeadm-bootstrap", "kubeadm-control-plane"],
+  "kind_cluster_name": "kind"
```

Comment:
Since the `kind_cluster_name` would affect the `allowed_contexts`, I'm wondering if we should create an issue to track trying to automate the updated of `allowed_contexts`, or if we should automatically append the proper context based on the `kind_cluster_name` (if provided)?

Reply:
I am pretty new to this so I think I am missing the relationship between kind_cluster_name and allowed_contexts.

Sounds like we could derive `allowed_contexts` from the `kind_cluster_name` somehow? Happy to add that in this PR if that makes more sense.

Reply:
Kind sets up a context in the kubeconfig for each cluster you create, the pattern is kind-<cluster name>, I believe. So, if you were to override the kind cluster name, you would also currently have to override the defined allowed_contexts to allow the correct context for the kind cluster you are using.

Reply:
We don't have to worry in this case since the apiserver will be on localhost and those are allowed by tilt implicitly. See: https://docs.tilt.dev/api.html#api.allow_k8s_contexts

But it might be worth noting in a comment perhaps?

Reply:
@chuckha when I was working on the Tilt setup, I had to put in `kind-kind` for allowed contexts, otherwise it wouldn't proceed. Not sure if something's changed since then, or my env is different? (Docker for Mac, kind 0.7)

Reply:
@detiber Gotcha! I totally get it now. One option that comes to mind is automatically appending `kind-${kind_cluster_name}` to the `allowed_contexts` list. 

Although it seems like we don't need to worry about it per @chuckha's comment?

Reply:
Ah, it looks like tilt support for detecting kind >= v0.6 requires tilt >= [v0.10.3](https://github.com/windmilleng/tilt/releases/tag/v0.10.23)

Reply:
I just tested this with the latest version of `tilt` by removing `kind-kind` from `allowed_contexts` and it worked as the docs suggest

Reply:
Ok, I must have tested with an older version of tilt, thanks!

Reply:
If we haven't already @alexbrand, maybe just add the version requirement to the docs.

Reply:
To be clear, we are looking to set the version requirements in the doc to:

Tilt >= v0.10.3

Thanks!

Reply:
And kind >= v0.16 (if it isn't already)

Reply:
Might be good, given those constraints to remove the setting of `kind-kind` in the example for `allowed_contexts` as well.

Reply:
Addressed. PTAL 👍

Reply:
Is it necessary to add it to the example, since we are basically just populating the default if unset?

Reply:
Good point. I'll remove it.